stl/inc/xhash

```diff
@@ -30,6 +30,13 @@ _STL_DISABLE_CLANG_WARNINGS
 #pragma push_macro("new")
 #undef new
 
+// Default to using floating point in xhash, defining a non-zero value will
+// avoid floating point arithmetic.
+#ifndef _XHASH_NO_FLOATING_POINT
+#define _XHASH_NO_FLOATING_POINT 0
+#endif // _XHASH_NO_FLOATING_POINT
+#pragma detect_mismatch("_XHASH_NO_FLOATING_POINT", _CRT_STRINGIZE(_XHASH_NO_FLOATING_POINT))

```

Comment:
No change requestd. It's unclear to me which stringization macro should be used, so I've opened #4314.